ga4gh/datamodel/reads.py

```diff
@@ -75,3 +76,80 @@ def toProtocolElement(self):
         readGroup.referenceSetId = None
         readGroup.sampleId = None
         return readGroup
+
+    def getReadAlignments(self, referenceName, referenceId, start, end):
+        """
+        Returns an iterator over the specified reads
+        """
+        # TODO use an index if it is found (accompanying .bai file)
```

Comment:
Ah, I didn't realise this. We'll just have to support only BAMs in that case (which is fine, less to worry about!). We can't support stuff without indexes, as these will be huge files in general. There's no way we can support retrieval by iteration and filtering, it's just not an option.